dipper/sources/GWASCatalog.py

```diff
@@ -263,69 +318,73 @@ def _process_haplotype(
-
-                query_result = so_ontology.query(so_query)
-
-                if len(list(query_result)) == 1:
-                    gene_id = DipperUtil.get_hgnc_id_from_symbol(mapped_genes[index])
-
-                    if gene_id is not None:
-                        geno.addAffectedLocus(snp_curie, gene_id)
-                        geno.addAffectedLocus(hap_id, gene_id)
-                        variant_in_gene_count += 1
-
-                gene_id = DipperUtil.get_hgnc_id_from_symbol(mapped_genes[index])
-                if gene_id is not None \
-                        and context_list[index] in ['upstream_gene_variant',
-                                                    'downstream_gene_variant']:
-                    graph.addTriple(
-                        snp_curie, self.resolve(context_list[index]), gene_id)
+        else:
 
-            else:
-                LOG.warning(
-                    "More mapped genes than snps, cannot disambiguate for %s",
-                    hap_label)
+            variant_in_gene_count = 0
+            for index, snp_curie in enumerate(snp_curies):
+                self._add_snp_to_graph(
+                    snp_curie, snp_labels[index], chrom_nums[index],
+                    chrom_positions[index], context_list[index])
+
+                if mapped_genes and len(mapped_genes) == len(snp_labels):
+                    so_class = self.resolve(context_list[index])
+                    so_query = """
+        SELECT ?variant_label
+        WHERE {{
+            {0} rdfs:subClassOf+ {1} ;
+            rdfs:label ?variant_label .
+        }}
+                    """.format(so_class, self.globaltt['gene_variant'])
+
+                    query_result = so_ontology.query(so_query)
+
+                    if len(list(query_result)) == 1:
+                        gene_id = DipperUtil.get_hgnc_id_from_symbol(
+                            mapped_genes[index])
+
+                        if gene_id is not None:
+                            geno.addAffectedLocus(snp_curie, gene_id)
+                            geno.addAffectedLocus(hap_id, gene_id)
+                            variant_in_gene_count += 1
 
-        # Seperate in case we want to apply a different relation
-        # If not this is redundant with triples added above
-        if len(mapped_genes) == variant_in_gene_count and len(set(mapped_genes)) == 1:
-            gene_id = DipperUtil.get_hgnc_id_from_symbol(mapped_genes[0])
-            geno.addAffectedLocus(hap_id, gene_id)
+                    gene_id = DipperUtil.get_hgnc_id_from_symbol(mapped_genes[index])
+                    if gene_id is not None \
+                            and context_list[index] in ['upstream_gene_variant',
+                                                        'downstream_gene_variant']:
+                            graph.addTriple(
+                                snp_curie, self.resolve(context_list[index]), gene_id)
+                    else:
```

Comment:
Over indented, I think this relates to the preceding if len(mapped_genes) == ...